### PR TITLE
Rearrange parser and response

### PR DESCRIPTION
**Pull request by ericdrosado**

### CODE REVIEW

**Review by cdemyanovich — APPROVED** [review]
Some good improvements here. Hopefully they lead to others.

File: Tests/Test/ResponseTest.swift [hunk 1]
```diff
@@ -3,8 +3,8 @@ import Server
 
 class ResponseTest: XCTestCase {
```
Comment (tommcgee) [inline Tests/Test/ResponseTest.swift:4]: Yay for more tests

File: Tests/Test/ResponseTest.swift [hunk 2]
```diff
@@ -15,56 +15,96 @@ class ResponseTest: XCTestCase {
...
+        let expectedResponse = "HTTP/1.1 418\r\nContent-Length: 12\r\nContent-type: text/html\r\n\r\nI'm a teapot"
+
+        let parsedRequest = parser.parseRequest(request: request)
+
+        XCTAssertEqual(expectedResponse, response.buildResponse(request: parsedRequest))
+    }
+
+    func testBuildResponseWillDecodeOperators() {
+        let request = "GET /parameters?variable_1=Operators%20%3C%2C%20%3E%2C%20%3D%2C%20!%3D%3B%20%2B%2C%20-%2C%20*%2C%20%26%2C%20%40%2C%20%23%2C%20%24%2C%20%5B%2C%20%5D%3A%20%22is%20that%20all%22%3F"
+        let expectedResponse = "HTTP/1.1 200 OK\r\nContent-Length: 127\r\nContent-type: text/html\r\n\r\n<!DOCTYPE html><html><body><h1>variable_1 = Operators <, >, =, !=; +, -, *, &, @, #, $, [, ]: \"is that all\"?</h1></body></html>"
```
Comment (tommcgee) [inline Tests/Test/ResponseTest.swift:104]: Don't have anything particular in mind, but I wonder if there is a way to clean up the `expectedResponse`s throughout this file.

**Review by tommcgee — APPROVED** [review]

File: Tests/Test/ResponseTest.swift (hunk 2 above)
Comment (cdemyanovich) [reply Tests/Test/ResponseTest.swift:104]: Some helper methods would do the trick. I was thinking of mentioning this when we next talked.

**Pull request merged**